### PR TITLE
Fix rest api access when using script/develop_and_serve

### PR DESCRIPTION
**Pull request by martetassyns**
## Proposed change

The script/develop_and_server only took into account UI components that connected through the websocket.
Now support has also been added for those that use the rest api.

## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

Tested through the System / Backup functionality

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion: https://github.com/home-assistant/developers.home-assistant/pull/2483
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -2,7 +2,8 @@
   "name": "Home Assistant Frontend",
   "build": {
     "dockerfile": "Dockerfile",
-    "context": ".."
+    "context": "..",
+    "options": [ "--add-host=host.docker.internal:host-gateway" ]
```
Comment (MindFreeze) [inline .devcontainer/devcontainer.json:6]:
```suggestion
    "options": ["--add-host=host.docker.internal:host-gateway"]
```

**Comment by bramkragten** [comment]
You create a proxy, but it is never used? All the API calls are still done directly to the external HA instance and not to the proxy.

You can test this easily with the config flows, as they use a REST API

**Comment by martetassyns** [comment]
@ bramkragten ok, I have confirmed that to use /config/integrations/integration/hacs you also need to configure cors. I will eventually look into seeing if we can improve this, but for now I just documented that it needs to be done. Can you agree that this is sufficient for this change? (as before this change it was also a requirement)

**Comment by martetassyns** [comment]
@bramkragten ok, ignore my earlier comment. I thought it would a lot of work to get the login flow to work properly when everything would be proxied through the frontend dev instance. But it appears it was rather easy. There is no longer a requirement to configure cors now.

**Comment by bramkragten** [comment]
> You create a proxy, but it is never used? All the API calls are still done directly to the external HA instance and not to the proxy.
> 
> [martetassyns/home-assistant-frontend@`b6ca770`/src/util/hass-call-api.ts#L58](https://github.com/martetassyns/home-assistant-frontend/blob/b6ca770f38e5e4a1563a02505b86a6c3dbe31811/src/util/hass-call-api.ts#L58)
> 
> You can test this easily with the config flows, as they use a REST API

I still don't see this addressed, the API calls are still done directly to the external host and not through the proxy.

Check config flows (add integration, integration options), automation editor, script editor, scene editor, etc

**Comment by martetassyns** [comment]
@bramkragten the change the fixes it is that we no longer override the HASS_URL when calling script/develop in script/develop_and_serve, all calls now go through the development frontend server.

File: script/serve.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+app.get("/auth/authorize", (req, res) => {
+  res.sendFile(path.join(repoDir, "hass_frontend/authorize.html"));
+});
+app.use("/auth", coreProxy);
+app.get("/onboarding", (req, res) => {
+  res.sendFile(path.join(repoDir, "hass_frontend/onboarding.html"));
+});
+app.get("*", (req, res) => {
+  res.sendFile(path.join(repoDir, "hass_frontend/index.html"));
+});
```
[the remark this answers is not in the document]
Comment (bramkragten) [reply script/serve.js:52]: If we have to change the index, we should do it inside the template with a flag, also, why does it use an interval?

[the remark this answers is not in the document]
Comment (MindFreeze) [reply script/serve.js:52]: Interval was the quickest dirty solution to test & show what needs to happen as `hassConnection` is not initialized when this script tag runs

**Comment by bramkragten** [comment]
We have done some more thinking and found a different way easier solution, setting allowed cors origins on the core instance 🙈 

https://www.home-assistant.io/integrations/http/#cors_allowed_origins

So this PR is no longer needed, we just need to document to set the above for `http://localhost:8124`

**Comment by martetassyns** [comment]
@bramkragten I disagree that having a proxy for the connections is not a benefit., it is still required because not all functionality uses the value from the login session to determine which /api/ to contact. For example the System/ Backup functionality (which connects directly to /api/... relative paths.

File: script/serve.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (martetassyns) [reply script/serve.js:52]: @MindFreeze why was this extension needed? it worked just fine with proxying everything?

**Comment by MindFreeze** [comment]
Setting CORS just works. A proxy is very complicated and very hacky. It's essentially a man in the middle attack. Also requires new dependancies

**Comment by martetassyns** [comment]
@MindFreeze @bramkragten Does that mean that every part of the code that uses /api/... as a relative path is wrong and should be reworked to use auth.data.hassUrl instead? (I've run into at least one, I don't mind fixing it that way)

**Pull request closed without merging**